### PR TITLE
argtable3: modernize + new version

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **argtable3/all**

New version + modernize

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
Do I understand correctly that solving https://github.com/conan-io/hooks/pull/320 means removing tools.os_info and platform? (I'm asking because this PR does not seem to do either)

**Comment by madebr** [comment]
@ericLemanissier 
https://github.com/conan-io/hooks/pull/320 does not forbid `tools.os_info` in all cases.
Only in those methods that should be build system independent.
You can still use `tools.os_info` in `system_requirements`. It's needed there to detect the Linux package manager.

**Comment by ericLemanissier** [comment]
yes, I get it. My question is: how is this PR solving conan-io/hooks#320 if it does not remove calls to `tools.os_info` or `platform` ?

**Comment by madebr** [comment]
@ericLemanissier 
Oh I see.
I used the script at https://github.com/conan-io/hooks/pull/320#issuecomment-903851889 to fix the issues (as a first pass).
It removed a final endline which caused git to trigger a diff which on its turn triggered me to look at argtable3.

**Comment by ericLemanissier** [comment]
OK, do maybe you could remove the reference to conan-io/hooks#320cin this PR 's description?

**Comment by madebr** [comment]
@ericLemanissier
Ok Done!

File: recipes/argtable3/all/conanfile.py [hunk 1]
```diff
@@ -30,6 +31,10 @@ class Argtable3Conan(ConanFile):
     def _source_subfolder(self):
         return "source_subfolder"
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
     def config_options(self):
```
Comment (madebr) [inline recipes/argtable3/all/conanfile.py:38]:
```suggestion
        return "source_subfolder"

    def config_options(self):
```

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — DISMISSED** [review]

File: recipes/argtable3/all/conanfile.py [hunk 2]
```diff
@@ -88,22 +89,28 @@ def package(self):
...
             {target_name: "argtable3::argtable3"}
         )
 
     def package_info(self):
-        self.cpp_info.libs = ["argtable3" if self.options.shared else "argtable3_static"]
+        suffix = ""
+        if tools.Version(self.version) >= "3.2.1":
+            if self.settings.build_type == "Debug":
+                suffix = "d"
+        self.cpp_info.libs = [("argtable3" if self.options.shared else "argtable3_static") + suffix]
```
Comment (SpaceIm) [inline recipes/argtable3/all/conanfile.py:104]:
```suggestion
        if not self.options.shared:
            suffix += "_static"
        if tools.Version(self.version) >= "3.2.1":
            if self.settings.build_type == "Debug":
                suffix += "d"
        self.cpp_info.libs = ["argtable3{}".format(suffix)]
```

Maybe simplier like this?

**Review by SpaceIm — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**